src/main/scala/mesosphere/marathon/core/volume/Volumes.scala

```diff
@@ -0,0 +1,97 @@
+package mesosphere.marathon.core.volume
+
+import com.wix.accord._
+import com.wix.accord.combinators.Fail
+import com.wix.accord.dsl._
+import com.wix.accord.Validator
+import com.wix.accord.ViolationBuilder._
+import mesosphere.marathon.WrongConfigurationException
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.AppDefinition
+import mesosphere.marathon.state.Volume
+import mesosphere.marathon.state.PersistentVolume
+
+/**
+  * Volumes is an interface implemented by storage volume providers
+  */
+sealed trait Volumes[T <: Volume] {
```

Comment:
Feels like there is a strong (1:1 ?) relation between specific `Volume` and specific `Volumes`.
Would it make sense to make the logic here part of `Volume`?